### PR TITLE
Decoupling Authentication Flow: Mark I

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're taking a first step towards encapsulating all of the WordPress Authentication-Y code into it's own framework:

- Renamed **SigninHelpers** >> **WordPressAuthenticator**.
- Implemented a mechanism to relay back to the host app all of the internal events being tracked by means of Tracks. **WordPressAuthenticator** will **not** talk to Tracks, directly.
- Implemented **WordPressAuthenticationTracker**, which will (always) live in the WPiOS side of things. This class is a simple wrapper, which will be in charge of listening Flow-Y events, and relay those events back to Tracks.

### To test:
1. Verify that the unit tests are still happy
2. Run a smoke test over the login flow, verify nothing breaks violently

Needs Review: @nheagy 
cc @astralbodies 

Sir, may i bug you with a hopefully fun review?
Thanks in advance!

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationTracker.swift [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import Foundation
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationTracker.swift]: No need to import Foundation here

File: WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift [hunk 2]
```diff
@@ -49,7 +49,7 @@ class Login2FAViewController: LoginViewController, NUXKeyboardResponder, UITextF
         nc.addObserver(self, selector: #selector(applicationBecameInactive), name: .UIApplicationWillResignActive, object: nil)
         nc.addObserver(self, selector: #selector(applicationBecameActive), name: .UIApplicationDidBecomeActive, object: nil)
 
-        WPAppAnalytics.track(.loginTwoFactorFormViewed)
+        WordPressAuthenticator.emit(event: .loginTwoFactorFormViewed)
```
[the remark this answers is not in the document]
Comment (jleandroperez) [reply WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift]: I'd rather *track*, but WordPressAuthenticator actually posts that event, but won't take care of tracking anything. On the Client App side, we'd setup a listener that would acutally *track* it.

Goal is to encapsulate this in a library, and cut the ties with WPAppAnalytics WPAnalytics.

File: WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationTracker.swift (hunk 1 above)
Comment (jleandroperez) [reply WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationTracker.swift]: Nice catch! removing!

File: WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (nheagy) [reply WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift]: I'd like `post` better than `emit`. The latter sounds a bit non-specific, and flatulence-related 😜

**Comment by jleandroperez** [comment]
@nheagy `.emit` renamed over to `.post` (thanks for the heads up, i had no idea about the _negative_ connotation).

Ready for another peek! thanks for your time Nate!!

**Comment by nheagy** [comment]
Ok, let's do this!

:shipit:

**Comment by jleandroperez** [comment]
@nheagy Thanks **A LOT** for your time!!!

**Pull request merged**